src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, NgZone, Renderer2} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {ClrDragEvent, ClrDragEventType} from "../interfaces/drag-event";
+import {ClrDragAndDropEventBus} from "./drag-and-drop-event-bus";
+
+@Injectable()
+export class ClrDragEventListener<T> {
+    private draggableEl: Node;
+
+    private listeners: (() => void)[];
+
+    private dragStart: Subject<ClrDragEvent<T>> = new Subject<ClrDragEvent<T>>();
+    private dragMove: Subject<ClrDragEvent<T>> = new Subject<ClrDragEvent<T>>();
+    private dragEnd: Subject<ClrDragEvent<T>> = new Subject<ClrDragEvent<T>>();
+
+    private hasDragStarted: boolean = false;
+
+    get dragStarted(): Observable<ClrDragEvent<T>> {
+        return this.dragStart.asObservable();
+    }
+
+    get dragMoved(): Observable<ClrDragEvent<T>> {
+        return this.dragMove.asObservable();
+    }
+
+    get dragEnded(): Observable<ClrDragEvent<T>> {
+        return this.dragEnd.asObservable();
+    }
+
+    constructor(private ngZone: NgZone, private renderer: Renderer2, private eventBus: ClrDragAndDropEventBus<T>) {}
+
+    public dragDataTransfer?: T;
+    public dragScope?: string|string[];
+
+    public attachDragListeners(draggableEl: Node) {
+        this.draggableEl = draggableEl;
+        this.listeners = [
+            this.customDragEvent(this.draggableEl, "mousedown", "mousemove", "mouseup"),
+            this.customDragEvent(this.draggableEl, "touchstart", "touchmove", "touchend")
+        ];
+    }
+
+    public detachDragListeners() {
+        if (this.listeners) {
+            this.listeners.map(event => event());
+        }
+    }
+
+    private customDragEvent(element: Node, startOnEvent: string, moveOnEvent: string, endOnEvent: string): () => void {
+        let moveListener: () => void;
+        let endListener: () => void;
+
+        return this.renderer.listen(element, startOnEvent, () => {
+            moveListener = this.ngZone.runOutsideAngular(() => {
+                return this.renderer.listen("document", moveOnEvent, (moveEvent: MouseEvent|TouchEvent) => {
```

Comment:
As mentioned before, I believe `renderer.listen` completely voids the `ngZone.runOutsideAngular`. Because it's "free" and doesn't impact anything, keep it like that for now and we can test for performance later on in the topic branch.